scripts/apps/search/directives/SearchParameters.js

```diff
@@ -28,7 +28,12 @@ export function SearchParameters($location, asset, tags, metadata, meta, common,
                 scope.query = params.q;
                 scope.flags = false;
                 scope.common = common;
-                scope.meta = meta = {};
+                scope.meta = meta;
+                for (var member in meta) {
```

Comment:
imo would be simpler if all that logic is in searchParameters, it could watch repo to trigger that setdefault func, so there is no need to split this logic it seems, or am I missing something?